src/event/entity/EntityExtinguishEvent.php

```diff
@@ -31,8 +31,21 @@
  * @phpstan-extends EntityEvent<Entity>
  */
 class EntityExtinguishEvent extends EntityEvent{
+	public const CAUSE_CUSTOM = 0;
+	public const CAUSE_WATER = 1;
+	public const CAUSE_WATER_CAULDRON = 2;
+	public const CAUSE_SPAWN = 3;
```

Comment:
`RESPAWN` would probably make more sense

Reply:
Although not implemented, powder snow and rain are missing.